src/app/georeferencer/qgsgcpcanvasitem.cpp

```diff
@@ -158,7 +160,25 @@ void QgsGCPCanvasItem::updatePosition()
     return;
   }
 
-  setPos( toCanvasCoordinates( mIsGCPSource ? mDataPoint->pixelCoords() : mDataPoint->mapCoords() ) );
+  if ( mIsGCPSource )
+  {
+    setPos( toCanvasCoordinates( mDataPoint->pixelCoords() ) );
+    return;
+  }
+  if ( mDataPoint->canvasCoords().isEmpty() )
+  {
+    QgsCoordinateReferenceSystem mapCrs = mMapCanvas->mapSettings().destinationCrs();
+    QgsPointXY mapCoords;
+    if ( mapCrs.toWkt() == mDataPoint->crs().toWkt() )
+      mapCoords = mDataPoint->mapCoords();
+    else
+    {
+      QgsCoordinateTransform transf( mDataPoint->crs(), mapCrs, QgsProject::instance() );
+      mapCoords = transf.transform( mDataPoint->mapCoords() );
+    }
```

Comment:
```suggestion
    QgsCoordinateTransform transf( mDataPoint->crs(), mapCrs, QgsProject::instance() );
    QgsPointXY mapCoords  = transf.transform( mDataPoint->mapCoords() );
```

The coordinate transform already short-cuts if source = dest, so we can simplify the logic here.